packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,226 @@
+  /// make sense i.e. spaceEvenly, spaceAround, spaceBetween, override the
+  /// value of gap based on the expected behavior of those alignments.
+  /// Otherwise, no-op.
+  void _gapOverride(double freeSpace) {
+    final gapOverridingAlignments = {
+      MainAxisAlignment.spaceEvenly,
+      MainAxisAlignment.spaceAround,
+      MainAxisAlignment.spaceBetween,
+    };
+    if (!gapOverridingAlignments.contains(mainAxisAlignment)) {
+      // mainAxisAlignment is not an alignment that can override gaps, so no-op.
+      return;
+    }
+
+    final int numberOfGaps;
+    switch (mainAxisAlignment) {
+      case MainAxisAlignment.spaceEvenly:
+        numberOfGaps = children.length + 1;
+        break;
+      case MainAxisAlignment.spaceAround:
+        numberOfGaps = children.length;
+        break;
+      case MainAxisAlignment.spaceBetween:
+        numberOfGaps = children.length - 1;
+        break;
+      default:
+        // this should never happen because of
+        // the guard at the start of this method.
+        throw Exception('Unexpected call to _gapOverride');
+    }
+    _gap = freeSpace / numberOfGaps;
+  }
+
+  void _layoutMainAxisImpl({
+    required List<PositionComponent> components,
+
+    /// - Zero when [MainAxisAlignment.start].
+    /// - [LayoutComponent.size] when [MainAxisAlignment.end] and
+    /// [MainAxisAlignment.spaceBetween]
+    /// - Half of free space when [MainAxisAlignment.center]
+    /// - [LayoutComponent.gap] when [MainAxisAlignment.spaceEvenly]
+    /// - Half of [LayoutComponent.gap] when [MainAxisAlignment.spaceAround]
+    Offset initialOffset = Offset.zero,
+
+    /// true if laying out from the end (bottom/right)
+    bool reverse = false,
+  }) {
+    final componentList = reverse ? components.reversed : components;
+    for (final (index, component) in componentList.indexed) {
+      final prevChild = index > 0 ? componentList.elementAt(index - 1) : null;
+      final reference = prevChild == null
+          // Essentially the same as start, but gap is set.
+          ? initialOffset.toVector2()
+          // The "end" at any loop other than the first is the previous
+          // child's top left position minus the gap.
+          : prevChild.topLeftPosition +
+              (reverse ? -Vector2.all(gap) : prevChild.size + Vector2.all(gap));
+      final positionOffset = reverse ? component.size : Vector2.zero();
+      final newPosition = Vector2.zero();
+      newPosition[mainAxisVectorIndex] =
+          (reference - positionOffset)[mainAxisVectorIndex];
+      component.topLeftPosition.setFrom(newPosition);
+    }
+  }
+
+  void _layoutCrossAxis() {
+    final components = children.whereType<PositionComponent>().toList();
+
+    // There is no need to track index because cross axis positioning is
+    // not influenced by sibling components.
+    for (final component in components) {
+      final newPosition = Vector2.copy(component.topLeftPosition);
```

Comment:
Use a static vector and do `setFrom` instead everywhere where you do `copy`, to avoid creating new vector objects.